Could not compile dependency :simdjsone

We're trying to pull this into a project but are getting compilation errors with github actions. It worked fine locally.

With the action:
`runs-on: ubuntu-latest`

And running:
```
erlang 27.1.2
elixir 1.17.3-otp-27
```

We get:
```
make: Entering directory '/home/runner/work/***/***/deps/simdjsone/c_src'
g++ -DSIMDJSON_DISABLE_DEPRECATED_API -fPIC -finline-functions -Wall -std=c++20 -O3 -DNDEBUG -I/home/runner/work/_temp/.setup-beam/otp/erts-15.1.2/include -I/home/runner/work/_temp/.setup-beam/otp/lib/erl_interface-5.5.2/include  -c -o simdjson.o simdjson.cpp
g++ -DSIMDJSON_DISABLE_DEPRECATED_API -fPIC -finline-functions -Wall -std=c++20 -O3 -DNDEBUG -I/home/runner/work/_temp/.setup-beam/otp/erts-15.1.2/include -I/home/runner/work/_temp/.setup-beam/otp/lib/erl_interface-5.5.2/include  -c -o simdjson_nif.o simdjson_nif.cpp
g++ simdjson.o simdjson_nif.o -L/home/runner/work/_temp/.setup-beam/otp/lib/erl_interface-5.5.2/lib -lei -shared -o /home/runner/work/***/***/_build/test/lib/simdjsone/priv/simdjsone.so
/usr/bin/ld: cannot open output file /home/runner/work/***/***/_build/test/lib/simdjsone/priv/simdjsone.so: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:76: /home/runner/work/***/***/_build/test/lib/simdjsone/priv/simdjsone.so] Error 1
make: Leaving directory '/home/runner/work/***/***/deps/simdjsone/c_src'
===> Hook for compile failed!
```

Noting that `***/***` is simply the replaced project name.

Update:

I can actually reproduce this locally as well. The error will occur the first attempt at `mix compile` when the `_build` directory is empty. The second `mix compile` runs without issue. Is there a timing issue where `simdjsone.so` isnt available when its expected to be?

I got the similar issue 
g++ -DSIMDJSON_DISABLE_DEPRECATED_API -fPIC -finline-functions -Wall -std=c++20 -O3 -DNDEBUG -I/usr/lib/erlang/erts-12.0/include -I/usr/lib/erlang/lib/erl_interface-5.0/include  -c -o simdjson.o simdjson.cpp
g++: error: unrecognized command line option ‘-std=c++20’; did you mean ‘-std=c++2a’?
make: *** [Makefile:81: simdjson.o] Error 1
make: Leaving directory '/home/hiteshv/omega/_build/default/lib/simdjsone/c_src'


changed -std=c++20’; to ‘-std=c++2a into make file and it worked

--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -71,11 +71,11 @@ info:
 simdjson.h simdjson.cpp:
 	wget -q https://raw.githubusercontent.com/simdjson/simdjson/master/singleheader/$@
 
-$(SO_OUTPUT): $(BASEDIR)/priv $(OBJECTS)
+$(SO_OUTPUT): $(PRIV_DIR) $(OBJECTS)
 	$(CXX) $(OBJECTS) $(LDFLAGS) -o $(SO_OUTPUT)
 
-$(BASEDIR)/priv:
-	@mkdir -p $(BASEDIR)/priv/
+$(PRIV_DIR):
+	@mkdir -p $(PRIV_DIR)
 
 simdjson.o: simdjson.cpp simdjson.h
 	$(COMPILE_CPP) -c $(OUTPUT_OPTION) $<

